src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -294,6 +294,83 @@ else if (value.contains(ResourceHandler.RESOURCE_IDENTIFIER)) {
             return context.getExternalContext().encodeResourceURL(url);
         }
     }
+    
+    /**
+     * Generate an <code>href</code> URL considering a base URL and some other params.
+     * This method consider existing query string and fragment in the base URL.
+     * @param baseUrl
+     * @param params
+     * @return 
```

Comment:
Since you have the javadoc here can you fill out the comment values for the params and return value?

Reply:
Sure. I've completed the javadoc for the method.